Makefile

```diff
@@ -46,12 +48,25 @@ lint:
 
 # Run tests
 ENVTEST_ASSETS_DIR = $(shell pwd)/testbin
-test: generate fmt vet manifests
+setup-envtest:
 	mkdir -p $(ENVTEST_ASSETS_DIR); \
-	test -f $(ENVTEST_ASSETS_DIR)/setup-envtest.sh || curl -sSLo $(ENVTEST_ASSETS_DIR)/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh; \
-	source $(ENVTEST_ASSETS_DIR)/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); \
+    test -f $(ENVTEST_ASSETS_DIR)/setup-envtest.sh || curl -sSLo $(ENVTEST_ASSETS_DIR)/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh; \
+    source $(ENVTEST_ASSETS_DIR)/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR)
+
+test: generate fmt vet manifests setup-envtest
+	go get golang.org/x/tools/cmd/cover; \
+ 	NOOBAA_CORE_IMAGE={NOOBAA_CORE_IMAGE} NOOBAA_DB_IMAGE={NOOBAA_DB_IMAGE} go test -v -coverprofile coverage.out ./...; \
+	$(MAKE) -f $(MAKEFILE) coverage
+
+unit-test: generate fmt vet manifests
+	go get golang.org/x/tools/cmd/cover; \
+	NOOBAA_CORE_IMAGE={NOOBAA_CORE_IMAGE} NOOBAA_DB_IMAGE={NOOBAA_DB_IMAGE} go test -v -coverprofile coverage.out ./controllers; \
```

Comment:
unit test only for controllers folder? It will block us achieving maximum coverage

Reply:
We can add the folders in our repository as we go along, while making sure that `tests` is excluded. IIRC right now only `./controllers` has unit tests.

Reply:
You could also just do `$(go list ./... | grep -v tests)` or some equivalent.